From 13bc9aaadd1020cd6e88cb6d98af221a476fab3a Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Fri, 12 Sep 2014 14:35:35 -0400
Subject: [redhat] kernel.spec: Add perf traceevent plugins installation

Message-id: <1410532537-3812-3-git-send-email-jolsa@redhat.com>
Patchwork-id: 94043
O-Subject: [PATCH RHEL7.1 BZ879133 2/4] spec: Add perf traceevent plugins installation
Bugzilla: 879133
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Petr Holasek <pholasek@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 879133
https://bugzilla.redhat.com/show_bug.cgi?id=879133

upstream
========
N/A - RHEL7 specific

description
===========
Adding installation of perf traceevent plugin libraries,
which handles quirks in tracepoint format parsing.

Plugins get installed under stadard lib directory:

  # rpm -ql perf | grep plugin
  /usr/lib64/traceevent/plugins
  /usr/lib64/traceevent/plugins/plugin_cfg80211.so
  /usr/lib64/traceevent/plugins/plugin_function.so
  /usr/lib64/traceevent/plugins/plugin_hrtimer.so
  /usr/lib64/traceevent/plugins/plugin_jbd2.so
  /usr/lib64/traceevent/plugins/plugin_kmem.so
  /usr/lib64/traceevent/plugins/plugin_kvm.so
  /usr/lib64/traceevent/plugins/plugin_mac80211.so
  /usr/lib64/traceevent/plugins/plugin_sched_switch.so
  /usr/lib64/traceevent/plugins/plugin_scsi.so
  /usr/lib64/traceevent/plugins/plugin_xen.so

Plugins get loaded for any tracepoint report:

  # perf record -e 'sched:*' -a
  ^C[ perf record: Woken up 1 times to write data ]
  [ perf record: Captured and wrote 0.160 MB perf.data (~6974 samples) ]

  # perf report -v --stdio | head
  registering plugin: /usr/lib64/traceevent/plugins/plugin_cfg80211.so
  registering plugin: /usr/lib64/traceevent/plugins/plugin_function.so
  registering plugin: /usr/lib64/traceevent/plugins/plugin_hrtimer.so
  registering plugin: /usr/lib64/traceevent/plugins/plugin_jbd2.so
  registering plugin: /usr/lib64/traceevent/plugins/plugin_kmem.so
  registering plugin: /usr/lib64/traceevent/plugins/plugin_kvm.so
  registering plugin: /usr/lib64/traceevent/plugins/plugin_mac80211.so
  registering plugin: /usr/lib64/traceevent/plugins/plugin_sched_switch.so
  registering plugin: /usr/lib64/traceevent/plugins/plugin_scsi.so
  registering plugin: /usr/lib64/traceevent/plugins/plugin_xen.so
  overriding event (311) sched:sched_wakeup with new print handler
  overriding event (310) sched:sched_wakeup_new with new print handler
  overriding event (309) sched:sched_switch with new print handler
  SNIP

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/redhat/kernel.spec.template b/redhat/kernel.spec.template
index 9241d3a..8ac2677 100644
--- a/redhat/kernel.spec.template
+++ b/redhat/kernel.spec.template
@@ -1016,7 +1016,7 @@ BuildKernel %make_target %kernel_image debug
 BuildKernel %make_target %kernel_image kdump
 %endif
 
-%global perf_make make %{?_smp_mflags} -C tools/perf -s V=1 WERROR=0 NO_LIBUNWIND=1 HAVE_CPLUS_DEMANGLE=1 NO_GTK2=1 NO_STRLCPY=1 prefix=%{_prefix}
+%global perf_make make %{?_smp_mflags} -C tools/perf -s V=1 WERROR=0 NO_LIBUNWIND=1 HAVE_CPLUS_DEMANGLE=1 NO_GTK2=1 NO_STRLCPY=1 prefix=%{_prefix} lib=%{_lib}
 %if %{with_perf}
 # perf
 %{perf_make} all
@@ -1177,7 +1177,7 @@ tar xjvf %{SOURCE23} -C $INSTALL_KABI_PATH
 
 %if %{with_perf}
 # perf tool binary and supporting scripts/binaries
-%{perf_make} DESTDIR=$RPM_BUILD_ROOT install-bin
+%{perf_make} DESTDIR=$RPM_BUILD_ROOT install
 # remove the 'trace' symlink.
 rm -f $RPM_BUILD_ROOT/%{_bindir}/trace
 
@@ -1379,6 +1379,7 @@ fi
 %{_bindir}/perf
 %dir %{_libexecdir}/perf-core
 %{_libexecdir}/perf-core/*
+%{_libdir}/traceevent
 %{_mandir}/man[1-8]/perf*
 %{_sysconfdir}/bash_completion.d/perf
 
-- 
1.7.1